NodeJS: Add descriptive link text

guskirb opened the pull request:
Made changes to the links in the Express 102, 103, 104 and 105 lessons.

<!-- Thank you for taking the time to contribute to The Odin Project. In order to get this pull request (PR) merged in a reasonable amount of time, you must complete this entire template. -->

## Because
<!-- Summarize the purpose or reasons for this PR, e.g. what problem it solves or what benefit it provides. -->
Provide descriptive link text.

## This PR
<!-- A bullet point list of one or more items describing the specific changes. -->
* Add descriptive link text to Express 102 lesson.
* Add descriptive link text to Express 103 lesson.
* Add descriptive link text to Express 104 lesson.
* Add descriptive link text to Express 105 lesson.

## Issue
<!--
If this PR closes an open issue in this repo, replace the XXXXX below with the issue number, e.g. Closes #2013.

If this PR closes an open issue in another TOP repo, replace the #XXXXX with the URL of the issue, e.g. Closes https://github.com/TheOdinProject/curriculum/issues/XXXXX

If this PR does not close, but is related to another issue or PR, you can link it as above without the 'Closes' keyword, e.g. 'Related to #2013'.

_Note:_ any pull request created for an issue that already has someone else assigned **will be closed without review**.
-->
Related to #27819

## Additional Information
<!-- Any other information about this PR, such as a link to a Discord discussion. -->


## Pull Request Requirements
<!-- Replace the whitespace between the square brackets with an 'x', e.g. [x]. After you create the PR, they will become checkboxes that you can click on. -->
-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

MaoShizhong reviewed (changes requested):
Thanks for sorting out these links, @guskirb!

Would you be good to fix up the remaining lint errors? If you follow the [curriculum repo contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md), it'll show you how to run lint checks and fixes locally.

Most of the lint errors should be fixable by running the correct script on a file (`project_express_105` will need the project scripts, not the lesson scripts). There will be a couple that can't be auto-fixed so will need manually fixing (the error message should be somewhat self explanatory).

For the "Required heading structure" error in `express_103_crud_and_mvc.md` line 62, that should be fixed by removing the `s` from `Knowledge checks` to make `Knowledge check`.

You can ignore the "Required heading structure" error for `express_104_view_templates.md`.

Let me know if you run into any issues with this.

guskirb commented:
> Small side note - I suspect you may have a formatter like Prettier that runs on save, as the `project 105` file had its `* *` emphasis (which is what we want) switched to `_ _` (which is not what we want). I'd recommend finding the keybinding for `Save without formatting` and getting used to using that when working on any external projects, as you can't guarantee they will use the same style guide as the formatter you have running automatically. Alternatively, you can have it so your formatter does not run on save but instead is bound to a separate keybinding. Then you just only use that when working on projects you know are fine to use your formatter with.

I've fixed the lint errors.

 I did have prettier set to reformat on save, which I've changed, thanks for letting me know.

MaoShizhong commented on nodeJS/express_and_mongoose/express_102_crud_and_mvc.md:
```diff
@@ -19,58 +19,58 @@ This section contains a general overview of topics that you will learn in this l
 CRUD is a concept that comes up a lot in web development, and it's the type of thing that might show up in interview questions so it's worth taking a little time to make sure you understand what it refers to.
 
-CRUD stands for: <span id="crud">_Create, Read, Update_ and _Delete_</span>. These are the four basic functions that you will be building into your database driven apps. If you are designing a CRUD interface that means that users can expect to be able to do these 4 things to items in the database (providing they have the appropriate permissions of course).
+CRUD stands for: <span id="crud">*Create, Read, Update* and *Delete*</span>. These are the four basic functions that you will be building into your database driven apps. If you are designing a CRUD interface that means that users can expect to be able to do these 4 things to items in the database (providing they have the appropriate permissions of course).
```
```suggestion
CRUD stands for: *Create, Read, Update* and *Delete*. These are the four basic functions that you will be building into your database driven apps. If you are designing a CRUD interface that means that users can expect to be able to do these 4 things to items in the database (providing they have the appropriate permissions of course).
```
Not something you've made but I just noticed that we have a redundant span here, as the `### CRUD` heading itself directly above already generates an `h3#crud`, so the span isn't actually needed for scroll either (since it's not like it's a long way away from the heading). Mind removing it while we're at it?

Everything else looks :man_cook: :pinched_fingers:

guskirb replied:
Done!

MaoShizhong reviewed (approved):
:rocket:
